### PR TITLE
add gzip docs support

### PR DESCRIPTION
**Pull request by machi1990**
-  Add Gzip support docs
- Also refactor the config to use a MemorySize configuration type for maxInput config knob.

### CODE REVIEW

File: docs/src/main/asciidoc/rest-json-guide.adoc [hunk 1]
```diff
@@ -484,6 +484,20 @@ If you want to use the `keycloak` Quarkus extension in your project, you should
...
+Quarkus comes with GZip support (even though it is not enabled by default). The following configuration knobs allow to configure GZip support.
+
+[source, properties]
+----
+quarkus.resteasy.gzip.enabled=true // <1>
+quarkus.resteasy.gzip.max-input=10M // <2>
+----
+
+<1> Enable Gzip support.
+<2> Configure the upper limit on deflated request body. The default value is `10M`.
```
Comment (emmanuelbernard) [inline docs/src/main/asciidoc/rest-json-guide.adoc]: Can you explain why one would put a upper limit? Why would I use a given property is important.

File: docs/src/main/asciidoc/rest-json-guide.adoc [hunk 2]
```diff
@@ -484,6 +484,20 @@ If you want to use the `keycloak` Quarkus extension in your project, you should
 configuration. See link:keycloak-guide#configuring-cors[the Keycloak guide] for more details.
 ====
 
+== GZip Support
```
Comment (emmanuelbernard) [inline docs/src/main/asciidoc/rest-json-guide.adoc]: Add space after a title 

```
== Foo

bar
```

**Review by emmanuelbernard — DISMISSED** [review]
Looks good except two minor comments

File: docs/src/main/asciidoc/rest-json-guide.adoc (hunk 1 above)
Comment (machi1990) [reply docs/src/main/asciidoc/rest-json-guide.adoc]: A little explanation added. WDYT?

File: docs/src/main/asciidoc/rest-json-guide.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/rest-json-guide.adoc]: Done

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java [hunk 3]
```diff
@@ -62,8 +62,8 @@
          * If the limit is exceeded, Resteasy will return Response
          * with status 413("Request Entity Too Large")
          */
-        @ConfigItem
-        public OptionalInt maxInput;
+        @ConfigItem(defaultValue = "10M")
```
Comment (n1hility) [inline extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java]: Out of curiosity, what lead you to picking 10meg as the default? I am not necessarily arguing for something different, I am just wondering if its mirroring a convention from elsewhere.

Comment (machi1990) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java]: I lifted it out of Resteasy. :-)

File: docs/src/main/asciidoc/rest-json-guide.adoc [hunk 4]
```diff
@@ -484,6 +484,22 @@ If you want to use the `keycloak` Quarkus extension in your project, you should
...
+
+[source, properties]
+----
+quarkus.resteasy.gzip.enabled=true // <1>
+quarkus.resteasy.gzip.max-input=10M // <2>
+----
+
+<1> Enable Gzip support.
+<2> Configure the upper limit on deflated request body. This is useful in limiting risks of attack related by packaging
+the attack in an entity that expands greatly. The default value is `10M`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-json-guide.adoc]: I would try something like "This is useful to mitigate potential attacks by limiting their reach."

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks great. I suggested a small change.

File: docs/src/main/asciidoc/rest-json-guide.adoc (hunk 4 above)
Comment (machi1990) [reply docs/src/main/asciidoc/rest-json-guide.adoc]: Thanks. Suggestion taken into account.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Let's wait for CI and merge. Thanks!

**Pull request merged**